spec/models/dataset_spec.rb

```diff
@@ -123,4 +123,19 @@
       expect(dataset.public_updated_at).to eq(dataset.updated_at)
     end
   end
+
+  describe "#released" do
+    it "returns true when there is a link" do
+      dataset = FactoryGirl.create(:dataset, datafiles: [FactoryGirl.create(:datafile)])
```

Comment:
How about using the `create_list` or `create_pair` method from FactoryGirl to make this a bit more expressive?

```rb
dataset = FactoryGirl.create(:dataset, datafiles: FactoryGirl.create_list(:datafile, 1))
```

See https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#building-or-creating-multiple-records

We could also maybe move this into the Factory definition using associations so we could have a `dataset_with_datafiles` factory – https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#associations

Reply:
The list technique is better, but I'm against having factories that aren't directly related to a single model - it reduces visibility too much for me.

Reply:
It makes it easier to focus on the detail of the test, and not the inconsequential detail of object creation.

Reply:
I think I'll compromise and do what I did with find - use a trait.